homeassistant/components/tankerkoenig/sensor.py

```diff
@@ -134,6 +133,16 @@ def unique_id(self) -> str:
         """Return a unique identifier for this entity."""
         return f"{self._station_id}_{self._fuel_type}"
 
+    @property
+    def device_info(self) -> DeviceInfo | None:
+        """Return device info."""
+        return DeviceInfo(
+            connections={(ATTR_ID, self._station_id)},
+            name=f"{self._city} {self._street} {self._house_number} {self._brand}",
+            manufacturer=self._brand,
```

Comment:
Are you sure that "manufacturer" is the right attribute? (or that the station's brand is the right value to set as manufacturer?)

Reply:
TBH, there is no really proper attribute in DeviceInfo, so the manufacture was the most matching one:
https://github.com/home-assistant/core/blob/574f4710aac2643ca616004ab6207290c5c17ec2/homeassistant/helpers/entity.py#L174-L190

result looks like:
![image](https://user-images.githubusercontent.com/35783820/159359881-241a5c35-2abb-4fb6-9c01-9477d6b1c89c.png)

Reply:
I think for this we need the input of some senior reviewer

Reply:
What do you think about this one?
![image](https://user-images.githubusercontent.com/35783820/159363722-c7e25ac6-916f-45f7-b9cb-e0b2b818fba7.png)

Reply:
No, this is the magic the the device detail view ^^ The devicename is omitted from the entity-names (just in this view)

device detail view
![image](https://user-images.githubusercontent.com/35783820/159365988-2825023c-1b50-469f-b049-0c84a1877830.png)

entities
![image](https://user-images.githubusercontent.com/35783820/159366034-c3fc3b89-8433-4fff-b4e2-6210885ec89c.png)